nav-app/src/app/matrix/technique-cell/technique-cell.component.ts

```diff
@@ -24,25 +24,25 @@ export class TechniqueCellComponent implements OnInit {
         if (this.showContextmenu) return false;
         if (this.viewModel.highlightedTechniques.size === 0) return false;
 
-        return (this.viewModel.highlightedTechniques.has(this.technique.id) && this.viewModel.highlightedTactic && this.viewModel.highlightedTactic.id == this.tactic.id);
+        return (this.viewModel.highlightedTechnique === this.technique && this.viewModel.highlightedTactic && this.viewModel.highlightedTactic.id === this.tactic.id);
     }
 
+
     public get isHighlighted(): boolean {
-        if (this.viewModel.selectTechniquesAcrossTactics) {
-            if (this.viewModel.selectSubtechniquesWithParent) {
-                return this.technique.isSubtechnique && this.viewModel.highlightedTechniques.has(this.technique.parent.id);
-            }
-        }
-        else if (this.viewModel.highlightedTactic && this.viewModel.highlightedTactic.id == this.tactic.id) {
-            if (this.viewModel.selectSubtechniquesWithParent) {
-                return this.technique.isSubtechnique && this.viewModel.highlightedTechniques.has(this.technique.parent.id);
+        let isHighlighted = this.showContextmenu;
+        let idToMatch = this.technique.id;
+        if (this.viewModel.selectSubtechniquesWithParent && this.technique.isSubtechnique) idToMatch = this.technique.parent.id;
+
+        if (this.viewModel.highlightedTechniques.has(idToMatch)) {
+            if (!this.viewModel.highlightedTactic) { // highlight is called from search component
+                return true;
+            } else if (this.viewModel.highlightedTactic) {
```

Comment:
Is there any functional difference between the `else if` used here and just an `else`? 
```
if (!this.viewModel.highlightedTactic) [...]
else if (this.viewModel.highlightedTactic)
```
Is the same as:
```
if (!this.viewModel.highlightedTactic) [...]
else
```
Because the `else if` is the inversion of the `if`.